source/MRMesh/MRVolumeSegment.cpp

```diff
@@ -211,7 +211,11 @@ tl::expected<VoxelBitSet, std::string> VolumeSegmenter::segmentVolume( float seg
     }
 
     // Segment volume
-    return segmentVolumeByGraphCut( volumePart_, segmentationExponentModifier, seedsInVolumePartSpace_[Inside], seedsInVolumePartSpace_[Outside] );
+    auto res = segmentVolumeByGraphCut( volumePart_, segmentationExponentModifier, seedsInVolumePartSpace_[Inside], seedsInVolumePartSpace_[Outside], cb );
+    if ( res.empty() )
+        return tl::make_unexpected( "Operation was cancelled" );
```

Comment:
can res be empty if operation was not cancelled? mb better return `tl::expected` from `segmentVolumeByGraphCut` to know for sure if it was OK or not (if empty res only return in case of cancelation then current behaviour is ok)